fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1546,32 +1547,36 @@ public void dropPartition(Database db, Table table, DropPartitionClause clause)
             }
         }
 
+        if (olapTable.isLakeTable()) {
+            List<Long> shardGroupIds = new ArrayList<>();
+            shardGroupIds.add(partition.getShardGroupId());
+            stateMgr.getStarOSAgent().deleteShardGroup(shardGroupIds);
+        }
+
         // log
         DropPartitionInfo info = new DropPartitionInfo(db.getId(), olapTable.getId(), partitionName, isTempPartition,
                 clause.isForceDrop());
         editLog.logDropPartition(info);
 
-        if (!tabletIdSet.isEmpty()) {
-            stateMgr.getShardManager().getShardDeleter().addUnusedShardId(tabletIdSet);
-        }
-
         LOG.info("succeed in droping partition[{}], is temp : {}, is force : {}", partitionName, isTempPartition,
                 clause.isForceDrop());
     }
 
     public void replayDropPartition(DropPartitionInfo info) {
         Database db = this.getDb(info.getDbId());
         db.writeLock();
-        Set<Long> tabletIdSet = new HashSet<Long>();
         try {
             OlapTable olapTable = (OlapTable) db.getTable(info.getTableId());
+            Partition partition = olapTable.getPartition(info.getPartitionName());
             if (info.isTempPartition()) {
                 olapTable.dropTempPartition(info.getPartitionName(), true);
             } else {
-                tabletIdSet = olapTable.dropPartition(info.getDbId(), info.getPartitionName(), info.isForceDrop());
+                olapTable.dropPartition(info.getDbId(), info.getPartitionName(), info.isForceDrop());
             }
-            if (!tabletIdSet.isEmpty()) {
-                stateMgr.getShardManager().getShardDeleter().addUnusedShardId(tabletIdSet);
+            if (olapTable.isLakeTable()) {
+                List<Long> shardGroupIds = new ArrayList<>();
+                shardGroupIds.add(partition.getShardGroupId());
+                stateMgr.getStarOSAgent().deleteShardGroup(shardGroupIds);
```

Comment:
will this cause FE to call starmgr delete again when doing editLOG replay?

Reply:
Yes, it will. I would remove this op here